alphagov/govuk-knowledge-graph-search
# Add redis session store

This PR adds Redis as a session store in the Express app to enable persistent user sessions.
Previously, user sessions were stored in memory, resulting in session loss upon server restart or revision changes.

All the changes happen in `app.ts`


# Changes Made
Added the ioredis library as a dependency to enable communication with the Redis instance.
Configured a Redis client in the Express app, establishing a connection to the Redis instance using the provided connection details.
Implemented the connect-redis middleware to create an Express-compatible session store using the Redis client.
Updated the session middleware configuration to use the Redis store for session management.


# Testing
You can follow these steps:

I have created a GCP Compute Engine "micro" instance solely for testing purposes. I will delete it after testing has been done.
1. SSH into the instance.
2. Run the following command to check the content of the Redis store:

```
redis-cli -h <redis-store-ip>
KEYS *
```
This will show the keys present in the Redis store.


To test session persistence:
Delete all cookies related to gov.uk from your browser.
Delete the content of the Redis store: `flushall`
Refresh the dev environment and log in.
Verify that a session has been created in the Redis store.
Refresh the page in the development environment, and ensure that you remain logged in.
Make a minor change on the branch (e.g., add a console.log statement), and deploy it as a new revision. Confirm that you are not logged out.
Delete your cookie, and log in again to create a new session in the Redis store.
Verify that the session ID in the cookie corresponds to the session stored in Redis.
